File: anika.dovob/src/main/java/com/example/anika/homework/chapter6/homework2/Member.java
package com.example.anika.homework.chapter6.homework2;

import jakarta.persistence.*;

@Entity(name = "Hw2Member")
@Table(name = "member_hw2")
public class Member {

    @Id
    @GeneratedValue
    @Column(name = "member_id")
    private Long id;

    private String name;

    @OneToOne  // ← fetch = FetchType.LAZY 추가해서 지연 로딩 확인!
//    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "locker_id")
    private Locker locker;

    public Member() {
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Locker getLocker() {
        return locker;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setLocker(Locker locker) {
        this.locker = locker;
    }
}

File: joon.bug/src/main/kotlin/com/example/joonbug/chapter6/homework2/Member.kt
package com.example.joonbug.chapter6.homework2

import jakarta.persistence.*

/**
 * 숙제 2: 일대일 지연 로딩 - Member 엔티티
 *
 * Member가 FK(locker_id)를 보유하는 주 테이블
 *
 * 실습: fetch 속성을 토글하며 SQL 로그 비교
 * - EAGER(기본값): Member 조회 시 Locker도 즉시 조회
 * - LAZY: Locker 접근 시점에 조회
 */
@Entity(name = "Hw2Member")
@Table(name = "hw2_member")
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var name: String = "",

    /**
     * 일대일 관계 - Member가 FK 보유 (연관관계 주인)
     *
     * 실습: 아래 fetch를 토글하며 SQL 로그 비교
     * - 현재 상태 (EAGER): Member 조회 시 Locker JOIN 또는 즉시 SELECT
     * - LAZY로 변경: locker 접근 시점에 SELECT
     */
    @OneToOne
    // @OneToOne(fetch = FetchType.LAZY)  // <- 주석 해제하여 비교
    @JoinColumn(name = "locker_id")
    var locker: Locker? = null
)